aten/src/ATen/native/cpu/DistributionKernels.cpp

```diff
@@ -103,11 +103,88 @@ void bernoulli_scalar_kernel(const TensorBase &self, double p, c10::optional<Gen
 }
 #endif
 
-static void exponential_kernel(TensorIteratorBase& iter, double lambda, c10::optional<Generator> gen) {
+static void exponential_kernel_default(TensorIteratorBase& iter, double lambda, c10::optional<Generator> gen) {
   CPUGeneratorImpl* generator = get_generator_or_default<CPUGeneratorImpl>(gen, detail::getDefaultCPUGenerator());
   templates::cpu::exponential_kernel(iter, lambda, generator);
 }
 
+#if !AT_MKL_ENABLED()
+void exponential_kernel(TensorIteratorBase& iter, double lambda, c10::optional<Generator> gen) {
+  exponential_kernel_default(iter, lambda, gen);
+}
+#else
+void exponential_kernel(TensorIteratorBase &iter, double lambda, c10::optional<Generator> gen) {
+  Tensor self = iter.tensor(0);
+  if (lambda > 0 && !std::isinf(lambda) && !std::isnan(lambda) && cpuinfo_initialize() &&
+      cpuinfo_vendor_intel == cpuinfo_get_processor(0)->core->vendor) {
```

Comment:
Can somebody please explain, why this check is necessary?

Reply:
@malfet Thanks for the ask. The same check was also applied to the `bernoulli_scalar_kernel` function (maybe added by Meta engineers in the past): https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cpu/DistributionKernels.cpp#L50

I think the check was added because the code path leverages MKL to optimize the random distribution generation and MKL is optimized for Intel CPUs:
https://www.intel.com/content/www/us/en/developer/articles/system-requirements/oneapi-math-kernel-library-system-requirements.html

Reply:
@jgong5 Thank you for your detailed explanation. Recently, I was occupied by other tasks and didn't reply in time.